### PR TITLE
service/cloudhsmv2: Modernize and refactor resources and datasource

### LINKED ISSUE #18591: tech-debt/cloudhsmv2: Add disappears and `tags` tests to resources and rename files to match resource prefix
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

* Acceptance tests should include their respective `_disappears` tests in addition to `tags` testing as they are supported by the resource
* the filenames (maybe documentation?) do not match the actual resource name prefixes and should be updated

### New or Affected Resource(s)

* aws_cloudhsm_v2_cluster
* aws_cloudhsm_v2_hsm

### References

* https://github.com/hashicorp/terraform-provider-aws/pull/18580#issuecomment-814237564
* https://github.com/hashicorp/terraform-provider-aws/issues/13826

### PR DESCRIPTION
**Pull request by bflad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #18591

Changes:

* Fix filenames to match resource naming
* Refactor to use finder and waiter package patterns
* Refactor internal failure retry handling to service client level
* Additional acceptance testing and removed resource handling

Output from acceptance testing (reduced concurrency to 5 cluster limit):

```
--- PASS: TestAccAWSCloudHsmV2Cluster_basic (254.16s)
--- PASS: TestAccAWSCloudHsmV2Cluster_disappears (285.53s)
--- PASS: TestAccAWSCloudHsmV2Cluster_Tags (322.03s)

--- PASS: TestAccAWSCloudHsmV2Hsm_AvailabilityZone (831.81s)
--- PASS: TestAccAWSCloudHsmV2Hsm_basic (898.92s)
--- PASS: TestAccAWSCloudHsmV2Hsm_disappears (856.53s)
--- PASS: TestAccAWSCloudHsmV2Hsm_disappears_Cluster (903.49s)
--- PASS: TestAccAWSCloudHsmV2Hsm_IpAddress (861.12s)
```

Output from sweeper in AWS Commercial:

```
2021/04/06 15:16:30 [DEBUG] Running Sweepers for region (us-west-2):
2021/04/06 15:16:30 [DEBUG] Running Sweeper (aws_cloudhsm_v2_cluster) in region (us-west-2)
2021/04/06 15:17:02 Sweeper Tests ran successfully:
        - aws_cloudhsm_v2_cluster
2021/04/06 15:17:02 [DEBUG] Running Sweepers for region (us-east-1):
2021/04/06 15:17:02 [DEBUG] Running Sweeper (aws_cloudhsm_v2_cluster) in region (us-east-1)
2021/04/06 15:17:03 Sweeper Tests ran successfully:
        - aws_cloudhsm_v2_cluster
ok      github.com/terraform-providers/terraform-provider-aws/aws       35.345s
```

### CODE REVIEW

File: aws/config.go [hunk 1]
```diff
@@ -685,6 +685,12 @@ func (c *Config) Client() (interface{}, error) {
 		}
 	})
 
+	client.cloudhsmv2conn.Handlers.Retry.PushBack(func(r *request.Request) {
```
Comment (YakDriver) [inline aws/config.go:688]: This is super cool!

File: aws/internal/service/cloudhsmv2/waiter/waiter.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	return nil, err
+}
+
+func ClusterDeleted(conn *cloudhsmv2.CloudHSMV2, id string, timeout time.Duration) (*cloudhsmv2.Cluster, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{cloudhsmv2.ClusterStateDeleteInProgress},
+		Target:     []string{cloudhsmv2.ClusterStateDeleted},
+		Refresh:    ClusterState(conn, id),
+		Timeout:    timeout,
+		MinTimeout: 30 * time.Second,
```
Comment (YakDriver) [inline aws/internal/service/cloudhsmv2/waiter/waiter.go:38]: I wonder if we shouldn't have these in more places.

**Review by YakDriver — APPROVED** [review]
So. Much. Clean. 🤩 

Acceptance tests on commercial:
```
--- PASS: TestAccAWSCloudHsmV2Cluster_basic (280.75s)
--- PASS: TestAccAWSCloudHsmV2Cluster_disappears (310.21s)
--- PASS: TestAccAWSCloudHsmV2Cluster_Tags (355.63s)

--- PASS: TestAccAWSCloudHsmV2Hsm_disappears (863.06s)
--- PASS: TestAccAWSCloudHsmV2Hsm_disappears_Cluster (872.69s)
--- PASS: TestAccAWSCloudHsmV2Hsm_IpAddress (947.21s)
--- PASS: TestAccAWSCloudHsmV2Hsm_basic (967.05s)
```

Acceptance tests on GovCloud:
```
--- PASS: TestAccAWSCloudHsmV2Cluster_disappears (270.73s)
--- PASS: TestAccAWSCloudHsmV2Cluster_Tags (290.92s)
--- PASS: TestAccAWSCloudHsmV2Cluster_basic (293.48s)

--- PASS: TestAccAWSCloudHsmV2Hsm_IpAddress (775.69s)
--- PASS: TestAccAWSCloudHsmV2Hsm_disappears_Cluster (866.53s)
--- PASS: TestAccAWSCloudHsmV2Hsm_AvailabilityZone (909.13s)
--- PASS: TestAccAWSCloudHsmV2Hsm_basic (939.57s)
```

**Comment by bflad** [comment]
Pulling this in, but aware I'll need to rebase the c* default tags PR.

**Pull request merged**